cli/config/config.go

```diff
@@ -0,0 +1,121 @@
+package config
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/nspcc-dev/neo-go/cli/cmdargs"
+	"github.com/nspcc-dev/neo-go/pkg/config"
+	"github.com/urfave/cli"
+	"gopkg.in/yaml.v3"
+)
+
+// NewCommands returns 'config' command.
+func NewCommands() []cli.Command {
+	return []cli.Command{
+		{
+			Name:  "config",
+			Usage: "NeoGo node configuration management",
+			Subcommands: []cli.Command{
+				{
+					Name:      "generate",
+					Usage:     "generate configuration files",
+					UsageText: "neo-go config generate [--privnet | --mainnet | --testnet | --unit_testnet | --mainnet_neofs | --testnet_neofs | --docker | --all] [--config-path path]",
+					Action:    configGenerate,
+					Flags: []cli.Flag{
+						cli.BoolFlag{Name: "privnet, p", Usage: "generate private network configuration"},
+						cli.BoolFlag{Name: "mainnet, m", Usage: "generate mainnet network configuration"},
+						cli.BoolFlag{Name: "testnet, t", Usage: "generate testnet network configuration"},
+						cli.BoolFlag{Name: "unit_testnet", Usage: "generate unit test network configuration"},
+						cli.BoolFlag{Name: "mainnet_neofs", Usage: "generate mainnet NeoFS network configuration"},
+						cli.BoolFlag{Name: "testnet_neofs", Usage: "generate testnet NeoFS network configuration"},
+						cli.BoolFlag{Name: "docker", Usage: "generate Docker configuration"},
+						cli.BoolFlag{Name: "all", Usage: "generate all networks configurations"},
+						cli.StringFlag{Name: "config-path", Usage: "path to the directory where configuration files will be generated"},
+					},
+				},
+			},
+		},
+	}
+}
+
+func configGenerate(ctx *cli.Context) error {
+	if err := cmdargs.EnsureNone(ctx); err != nil {
+		return err
+	}
+
+	filesToGenerate := make(map[string]config.Config)
+
+	if ctx.Bool("all") {
+		filesToGenerate = map[string]config.Config{
+			"protocol.mainnet.yml":               Mainnet,
+			"protocol.mainnet.neofs.yml":         MainnetNeoFS,
+			"protocol.testnet.yml":               Testnet,
+			"protocol.testnet.neofs.yml":         TestnetNeoFS,
+			"protocol.unit_testnet.yml":          UnitTestnet,
+			"protocol.unit_testnet.single.yml":   UnitTestnetSingle,
+			"protocol.privnet.docker.one.yml":    PrivnetDockerOne,
+			"protocol.privnet.docker.two.yml":    PrivnetDockerTwo,
+			"protocol.privnet.docker.three.yml":  PrivnetDockerThree,
+			"protocol.privnet.docker.four.yml":   PrivnetDockerFour,
+			"protocol.privnet.docker.single.yml": PrivnetDockerSingle,
+			"protocol.privnet.yml":               Privnet,
```

Comment:
These constants are duplicated several lines below. Can we move them under designated package-wide `const` definition?